File: android_app_p/app/src/main/java/com/comicviewer/data/model/Chapter.kt
package com.comicviewer.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Represents a single chapter within a manga.
 */
@Serializable
data class Chapter(
    val id: Int,

    @SerialName("manga_id")
    val mangaId: Int,

    @SerialName("chapter_number")
    val chapterNumber: Double,

    val title: String? = null,

    @SerialName("file_path")
    val filePath: String = "",

    @SerialName("page_count")
    val pageCount: Int = 0,

    @SerialName("created_at")
    val createdAt: String? = null
) {
    /**
     * Formatted chapter title for display.
     *
     * Examples: "Chapter 1", "Ch. 1.5 – Special Chapter"
     */
    val displayTitle: String
        get() {
            val numStr = if (chapterNumber % 1.0 == 0.0) {
                chapterNumber.toInt().toString()
            } else {
                String.format("%.1f", chapterNumber)
            }
            return if (!title.isNullOrBlank()) {
                "Ch. $numStr – $title"
            } else {
                "Chapter $numStr"
            }
        }
}

File: ios_app_p/ComicViewer/Models/Chapter.swift
// Chapter.swift
// Comic Viewer – Chapter Data Model
//
// Codable model representing a single chapter within a manga.

import Foundation

/// Represents a chapter (or volume) within a manga.
struct Chapter: Codable, Identifiable, Hashable {
    let id: Int
    let mangaId: Int
    let chapterNumber: Double
    let title: String?
    let filePath: String
    let pageCount: Int
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case mangaId = "manga_id"
        case chapterNumber = "chapter_number"
        case title
        case filePath = "file_path"
        case pageCount = "page_count"
        case createdAt = "created_at"
    }

    /// Formatted chapter number for display (e.g., "Ch. 1" or "Ch. 1.5")
    var displayTitle: String {
        let numStr = chapterNumber.truncatingRemainder(dividingBy: 1) == 0
            ? String(format: "%.0f", chapterNumber)
            : String(format: "%.1f", chapterNumber)

        if let title, !title.isEmpty {
            return "Ch. \(numStr) – \(title)"
        }
        return "Chapter \(numStr)"
    }
}
